examples/REST-API/README.md

```diff
@@ -0,0 +1,391 @@
+# Using the DIGITS REST API
+
+Table of Contents
+=================
+* [Introduction](#introduction)
+* [Image classification](#image-classification)
+    * [Creating the classification dataset](#creating-the-classification-dataset)
+    * [Creating the classification model](#creating-the-classification-model)
+    * [Classification](#classification)
+    * [Deleting a job](#deleting-a-job)
+* [Regression](#regression)
+    * [Creating the regression dataset](#creating-the-regression-dataset)
+    * [Creating the regression model](#creating-the-regression-model)
+    * [Inference](#inference)
+
+## Introduction
+
+The DIGITS REST API is a programming interface to DIGITS.
+In the spirit of [REST](https://en.wikipedia.org/wiki/Representational_state_transfer), the API may be used to make self-contained, stateless queries to DIGITS.
+In particular, the API may be used to create datasets and models, retrieve job information and perform inference on a trained model.
+Besides, this interface is easily scriptable, which allows for actions to be performed using DIGITS programmatically.
+
+In the first part of this walk-through we will see how the API may be used to create an image classification model.
+In the second part we will see how to create a regression model.
+
+> NOTE: this is not a comprehensive guide to the DIGITS REST API.
+> You should be able to refer to the code if in doubt about a specified feature.
+
+We will be using the `curl` command to interact with DIGITS from command line.
+You may use any URL access library to carry out the same actions from your favorite programming language.
+
+We will be assuming DIGITS is running on `localhost`.
+If DIGITS is running on another host/port, please adjust the command lines accordingly.
+
+## Image classification
+
+This is another way of running the [Getting started](../../docs/GettingStarted.md) tutorial.
+If you haven't gone through this tutorial, please do so now.
+
+### Creating the classification dataset
+
+In order to create a dataset, you will first need to log in.
```

Comment:
It's probably worth noting that you don't need to do this with a cookie. You can just add `-F username=luke` to the end of any `curl` command.

https://github.com/NVIDIA/DIGITS/blob/v4.0.0/digits/utils/auth.py#L11-L13